### PR TITLE
Display phpmodules versions on admin page

### PR DESCRIPTION
**Pull request by pb66**
See #660

### CODE REVIEW

**Comment by Paul-Reed** [comment]
Hi Paul, would it be possible to only print the module version number if one exists, instead of repeatedly printing the suffix ()?
version
Paul

**Comment by pb66** [comment]
I'm sure it is possible, but it's not something I know how to do off-hand. 

Technically it IS only printing the version number if one exists, it's the brackets that are not currently optional. It would require some sort of `if phpversion($val) is not null` then print the brackets and the version or to wrap the whole module name in brackets so you get (redis 2.2.8) when a version is available and (openssl) when there isn't (slightly neater perhaps). Otherwise without the brackets it just becomes a list of module names with a scattering of version numbers intermingled with them. It ideally needs to be clear what the version number belongs to by grouping the version no and module name.

Lets see what other comments are made before I spend too much time on it, if it's in but needs to be tidier I'm sure I or someone else will find a way to do so.

**Comment by Paul-Reed** [comment]
Agree
I wonder if the list would be clearer if the entries were comma separated. 

`ftp,   gettext,   SPL(0.2),   iconv,  mbstring,   session,   posix,  SimpleXML(0.1),  soap,   sockets,   Phar(2.0.2),`

...that would keep the module & version together.

Paul

**Comment by JonMurphy** [comment]
what about columns?!?

screen shot 2017-04-21 at 1 58 53 pm
(i am joking!!)

**Comment by pb66** [comment]
Comma or no commas it would still need the code to control the brackets unless you were to just use the comma in place of the brackets eg
```
ftp, gettext, SPL 0.2, iconv, mbstring, session, posix, SimpleXML 0.1, soap, sockets, Phar 2.0.2,
```
The additional "if exists" code will be needed when ever there is something other than the version printed ONLY when a version exists eg the brackets.

Just wrapping the name and version in brackets or separating with commas or any other char doesn't need any further coding.

I think there has already been a comma or no comma discussion around the modules list and the emoncms modules list too, I know there have been some quite strong feelings aired about the formatting of these lists.

**Comment by Paul-Reed** [comment]
That looks good to me Paul.
Much clearer, and no redundant brackets.

**Comment by glynhudson** [comment]
Sounds good, happy for this to be merged once everyone is happy 👍 

Thanks.

**Comment by JonMurphy** [comment]
I changed part of line 331 to `echo "$val(".phpversion($val).") &nbsp; ";` and it looks good to me!  Merge it!

**Comment by Paul-Reed** [comment]
Have you a screenshot @JonMurphy

**Comment by JonMurphy** [comment]
screen shot 2017-04-21 at 5 15 37 pm

**Comment by JonMurphy** [comment]
maybe I grabbed the wrong change?!?

**Comment by JonMurphy** [comment]
how about changing the php modules to this?

bcmath &nbsp; bz2 &nbsp; calendar &nbsp; Core v5.6.29-0+deb8u1 &nbsp; ctype &nbsp; curl &nbsp; date v5.6.29-0+deb8u1 &nbsp; dba &nbsp; dio v0.0.4RC4 &nbsp; dom v20031129 &nbsp; ereg &nbsp; exif v1.4 &nbsp; fileinfo v1.0.5 &nbsp; filter v0.11.0 &nbsp; ftp &nbsp; gettext &nbsp; hash v1.0 &nbsp; iconv &nbsp; json v1.3.6 &nbsp; libxml &nbsp; mbstring &nbsp; mcrypt &nbsp; mhash &nbsp; mosquitto v0.3.0 &nbsp; mysql v1.0 &nbsp; mysqli v0.1 &nbsp; openssl &nbsp; pcntl &nbsp; pcre &nbsp; PDO v1.0.4dev &nbsp; pdo_mysql v1.0.2 &nbsp; Phar v2.0.2 &nbsp; posix &nbsp; readline v5.6.29-0+deb8u1 &nbsp; redis v2.2.7 &nbsp; Reflection &nbsp; session &nbsp; shmop &nbsp; SimpleXML v0.1 &nbsp; soap &nbsp; sockets &nbsp; SPL v0.2 &nbsp; standard v5.6.29-0+deb8u1 &nbsp; sysvmsg &nbsp; sysvsem &nbsp; sysvshm &nbsp; tokenizer v0.1 &nbsp; wddx &nbsp; xml &nbsp; xmlreader v0.1 &nbsp; xmlwriter v0.1 &nbsp; Zend OPcache v7.0.6-devFE &nbsp; zip v1.12.5 &nbsp; zlib v2.0 &nbsp; 

The modules are in alphabetical order, round brackets removed, added "v" for version.  I did not add commas since it may have caused strong reactions.

I am not a PHP coder so there are probably better ways to do this:

```php
<?php
$array = get_loaded_extensions();
natcasesort($array);

while (list($key, $val) = each($array))
	{
	$ver = phpversion($val);
	echo $val;
	if (!empty($ver) && is_numeric($ver[0]))
		{
		$first = split(" ", $ver);
		echo " v" . $first[0];
		}

	echo " &nbsp; ";
	} ?>
```
this is not the final code that is added to the **admin_main_view.php** since I added `get_loaded_extensions()` command.

tl;dr;
**here is the current stable (master?) release of the PHP Modules output:**
Core &nbsp; date &nbsp; ereg &nbsp; libxml &nbsp; openssl &nbsp; pcre &nbsp; zlib &nbsp; bcmath &nbsp; bz2 &nbsp; calendar &nbsp; ctype &nbsp; dba &nbsp; dom &nbsp; hash &nbsp; fileinfo &nbsp; filter &nbsp; ftp &nbsp; gettext &nbsp; SPL &nbsp; iconv &nbsp; mbstring &nbsp; pcntl &nbsp; session &nbsp; posix &nbsp; Reflection &nbsp; standard &nbsp; shmop &nbsp; SimpleXML &nbsp; soap &nbsp; sockets &nbsp; Phar &nbsp; exif &nbsp; sysvmsg &nbsp; sysvsem &nbsp; sysvshm &nbsp; tokenizer &nbsp; wddx &nbsp; xml &nbsp; xmlreader &nbsp; xmlwriter &nbsp; zip &nbsp; PDO &nbsp; curl &nbsp; dio &nbsp; json &nbsp; mcrypt &nbsp; mosquitto &nbsp; mysql &nbsp; mysqli &nbsp; pdo_mysql &nbsp; readline &nbsp; redis &nbsp; mhash &nbsp; Zend OPcache &nbsp; 

**Here is what Taz Paul coded:**
Core(5.6.29-0+deb8u1) &nbsp; date(5.6.29-0+deb8u1) &nbsp; ereg() &nbsp; libxml() &nbsp; openssl() &nbsp; pcre() &nbsp; zlib(2.0) &nbsp; bcmath() &nbsp; bz2() &nbsp; calendar() &nbsp; ctype() &nbsp; dba() &nbsp; dom(20031129) &nbsp; hash(1.0) &nbsp; fileinfo(1.0.5) &nbsp; filter(0.11.0) &nbsp; ftp() &nbsp; gettext() &nbsp; SPL(0.2) &nbsp; iconv() &nbsp; mbstring() &nbsp; pcntl() &nbsp; session() &nbsp; posix() &nbsp; Reflection($Id: 5f15287237d5f78d75b19c26915aa7bd83dee8b8 $) &nbsp; standard(5.6.29-0+deb8u1) &nbsp; shmop() &nbsp; SimpleXML(0.1) &nbsp; soap() &nbsp; sockets() &nbsp; Phar(2.0.2) &nbsp; exif(1.4 $Id: 8b0e34c10dc8a04b8e81d9d79985b2566141b03d $) &nbsp; sysvmsg() &nbsp; sysvsem() &nbsp; sysvshm() &nbsp; tokenizer(0.1) &nbsp; wddx() &nbsp; xml() &nbsp; xmlreader(0.1) &nbsp; xmlwriter(0.1) &nbsp; zip(1.12.5) &nbsp; PDO(1.0.4dev) &nbsp; curl() &nbsp; dio(0.0.4RC4) &nbsp; json(1.3.6) &nbsp; mcrypt() &nbsp; mosquitto(0.3.0) &nbsp; mysql(1.0) &nbsp; mysqli(0.1) &nbsp; pdo_mysql(1.0.2) &nbsp; readline(5.6.29-0+deb8u1) &nbsp; redis(2.2.7) &nbsp; mhash() &nbsp; Zend OPcache(7.0.6-devFE) &nbsp;

**Comment by Paul-Reed** [comment]
Looks good to me Jon.

**Comment by TrystanLea** [comment]
Is this an older version of the pull request I just merged? or should I have merged this one first?

**Comment by Paul-Reed** [comment]
It's 2 versions of the same function. This PR was the original concept, but
we had been debating which version presented the info more clearly.
We had not merged the other PR because we wanted to ensure that it worked
OK on Windows first. But I'm sure we'll know soon enough!

**Comment by JonMurphy** [comment]
FWIW - I installed VirtualBox (on iMac) and installed Windows 7 and ran the change in #665 and all looks A-OK.  The **display phpmodules versions** changes look the same as a non-Windows machine (with the version number.)  Disclaimer - I am not a windows guru.  YMMV

**Comment by Paul-Reed** [comment]
Can this be closed now Paul? @pb66

**Comment by JonMurphy** [comment]
from my viewpoint: "yes"
